numpy/lib/polynomial.py

```diff
@@ -603,6 +603,8 @@ def polyfit(x, y, deg, rcond=None, full=False, w=None, cov=False):
         #  it is included here because the covariance of Multivariate Student-T
         #  (which is implied by a Bayesian uncertainty analysis) includes it.
         #  Plus, it gives a slightly more conservative estimate of uncertainty.
+        if (len(x) - order - 2.0) < 1:
+            raise ValueError("need at least three degrees of freedom for covariance matrix")
```

Comment:
The message isn't quite right. What is needed is that the number of data points must exceed the degree + 3.